### PR TITLE
Basic Flutter Error extension event handling.

### PR DESCRIPTION
**Pull request by pq**
Basic Flutter Error extension event handling.

image

As a next step we'll want to do something sensible w/ the parsed `DiagnosticsNode` pending UI exploration in the devtools version.  In the meantime this proves out inspector service initialization and interaction.

/cc @jacob314

### CODE REVIEW

File: src/io/flutter/logging/FlutterLogEntryParser.java [hunk 1]
```diff
@@ -83,25 +91,37 @@ public void onError(RPCError error) {
     }
   }
 
-  private FlutterDebugProcess debugProcess;
+  static class InspectorHelper {
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogEntryParser.java]: Is this helper class useful given when operating you typically need to handle only having a CompletableFuture<InspectorService.ObjectGroup> due to timing?

File: src/io/flutter/logging/FlutterLogEntryParser.java [hunk 2]
```diff
@@ -126,47 +146,53 @@ private FlutterLogEntry parseLoggingEvent(@NotNull Event event) {
...
-          entry.setMessage(((Instance)response).getValueAsString());
-          contentChangedDispatcher.getMulticaster().onContentUpdate();
-        }
-      });
+      if (debugProcess != null) {
+        final Isolate isolate = new Isolate(json.get("isolate").getAsJsonObject());
+        debugProcess.getVmServiceWrapper().getObject(isolate.getId(), message.getId(), new GetObjectAdapter() {
+          @Override
+          public void received(Obj response) {
+            entry.setMessage(((Instance)response).getValueAsString());
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogEntryParser.java]: not a new issue but isn't this on a background thread updating UI code?

File: src/io/flutter/logging/FlutterLogEntryParser.java [hunk 3]
```diff
@@ -223,19 +249,57 @@ private static long timestamp(Event event) {
...
+      // TODO(pq): process children: diagnosticsNode.getChildren().getNow(null) and set data to something structured.
+      entry.setData(json.toString());
+      entries.add(entry);
+    }
+    return entries;
+  }
+
+  @Nullable
+  private DiagnosticsNode parseDiagnosticsNode(@NotNull JsonObject json) {
+    if (inspectorObjectGroup == null) {
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogEntryParser.java]: how can this happen? Should this be an assert as it should generally indicate a logic bug.

File: src/io/flutter/logging/FlutterLogEntryParser.java [hunk 4]
```diff
@@ -223,19 +249,57 @@ private static long timestamp(Event event) {
...
+      FlutterUtils.warn(LOG, "Attempt to create a diagnostics node before the object group is set.");
+      return null;
+    }
+    if (app == null) {
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogEntryParser.java]: looks like an assert as well.

File: src/io/flutter/logging/FlutterLogEntryParser.java [hunk 5]
```diff
@@ -332,4 +396,9 @@ public FlutterLogEntry parseConsoleEvent(String text, ConsoleViewContentType typ
     // TODO(pq): handle else (errors, etc).
     return null;
   }
+
+  public void setupInspector(@NotNull FlutterApp app, @NotNull VmService vmService) {
```
Comment (jacob314) [inline src/io/flutter/logging/FlutterLogEntryParser.java]: Nit: might be nice to match the naming convention of similar methods that take an app and vmService to initialize debugger state?

File: src/io/flutter/logging/FlutterLogEntryParser.java (hunk 1 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogEntryParser.java]: Good point.  This class did more once but now doesn't have much reason to exist.

Removing. 👍

File: src/io/flutter/logging/FlutterLogEntryParser.java (hunk 2 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogEntryParser.java]: On my list to look into.  Thanks for the nudge.

File: src/io/flutter/logging/FlutterLogEntryParser.java (hunk 3 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogEntryParser.java]: Updated.

File: src/io/flutter/logging/FlutterLogEntryParser.java (hunk 4 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogEntryParser.java]: You're right.  Updated.

File: src/io/flutter/logging/FlutterLogEntryParser.java (hunk 5 above)
Comment (pq) [reply src/io/flutter/logging/FlutterLogEntryParser.java]: Sure thing.  Done.

**Pull request merged**